lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -2089,53 +2094,95 @@ diagnoseMatch(ModuleDecl *module, NormalProtocolConformance *conformance,
+    auto PBD = cast<VarDecl>(witness)->getParentPatternBinding();
+    diag.fixItReplace(PBD->getStartLoc(), getTokenText(tok::kw_var));
     break;
+  }
 
-  case MatchKind::PrefixNonPrefixConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness,
-                   diag::protocol_witness_prefix_postfix_conflict, false,
-                   match.Witness->getAttrs().hasAttribute<PostfixAttr>() ? 2
-                                                                         : 0);
+  case MatchKind::PrefixNonPrefixConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(
+        witness, diag::protocol_witness_prefix_postfix_conflict, false,
+        witness->getAttrs().hasAttribute<PostfixAttr>() ? 2 : 0);
+    if (witness->getAttrs().hasAttribute<PostfixAttr>()) {
+      auto attr = witness->getAttrs().getAttribute<PostfixAttr>();
+      diag.fixItReplace(attr->getLocation(), "prefix");
+    }
     break;
+  }
+
+  case MatchKind::PostfixNonPostfixConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(
+        witness, diag::protocol_witness_prefix_postfix_conflict, true,
+        witness->getAttrs().hasAttribute<PrefixAttr>() ? 1 : 0);
+    if (witness->getAttrs().hasAttribute<PrefixAttr>()) {
+      auto attr = witness->getAttrs().getAttribute<PrefixAttr>();
+      diag.fixItReplace(attr->getLocation(), "postfix");
+    }
 
-  case MatchKind::PostfixNonPostfixConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness,
-                   diag::protocol_witness_prefix_postfix_conflict, true,
-                   match.Witness->getAttrs().hasAttribute<PrefixAttr>() ? 1
-                                                                        : 0);
     break;
-  case MatchKind::MutatingConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness,
-                   diag::protocol_witness_mutation_modifier_conflict,
-                   SelfAccessKind::Mutating);
+  }
+  case MatchKind::MutatingConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(
+        witness, diag::protocol_witness_mutation_modifier_conflict,
+        SelfAccessKind::Mutating);
+    auto attr = witness->getAttrs().getAttribute<MutatingAttr>();
+    diag.fixItRemove(attr->getLocation());
+
     break;
-  case MatchKind::NonMutatingConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness,
-                   diag::protocol_witness_mutation_modifier_conflict,
-                   SelfAccessKind::NonMutating);
+  }
+  case MatchKind::NonMutatingConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(
+        witness, diag::protocol_witness_mutation_modifier_conflict,
+        SelfAccessKind::NonMutating);
+    auto attr = witness->getAttrs().getAttribute<NonMutatingAttr>();
+    diag.fixItRemove(attr->getLocation());
```

Comment:
*Definitely* questionable even though it's less likely to cause compilation errors. In practice, though, this conflict kind is unused because a non-mutating witness *can* satisfy a mutating requirement. It can be removed in a separate commit.